### PR TITLE
feat: support css variables for SvgCss

### PR DESCRIPTION
**Pull request by bohdanprog**
# Summary
Feature #2380 

We want to add support for CSS variables when passing them to parse the SVG XML source function.

## Test Plan
Test app -> src -> Test2380

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     		 |
| MacOS   |    ✅        |
| Android |    ✅     	|
| Web     |    ✅		    |

### CODE REVIEW

File: apps/test-examples/src/Test2380.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const xml = `
+  <svg width="32" height="32" id="mySVG">
+    <style>
+      #mySVG {
+        --my-color: #ff0000;
+        --my-width:150px;
+        --my-height:200px;
+      }
+    </style>
+    <rect fill="var(--my-color)" x="0" y="0" width="var(--my-width)" height="var(--my-height)" />
```
Comment (jakex7) [inline apps/test-examples/src/Test2380.tsx:13]: I believe `width` and `height` attributes can not be CSS variables. (or at least they don't work in chrome)

File: src/xml.tsx [hunk 2]
```diff
@@ -435,6 +438,40 @@ export function parse(source: string, middleware?: Middleware): JsxAST | null {
...
+          const variableName = value.slice(4, -1).trim();
+          style[camelProperty] = cssVariables[variableName] || value;
+        } else {
+          style[camelProperty] = value;
+        }
+      }
+    }
+    return style;
+  }
+
```
Comment (jakex7) [inline src/xml.tsx:474]: I don't think that it should be done here, as `SvgXml` does not support CSS styles at all.

**Review by jakex7 — COMMENTED** [review]
Left some comments

File: apps/test-examples/src/Test2380.tsx (hunk 1 above)
Comment (bohdanprog) [reply apps/test-examples/src/Test2380.tsx]: Yes, you're right. As I see it, it should support: fill, stroke, and color properties.

File: src/xml.tsx (hunk 2 above)
Comment (bohdanprog) [reply src/xml.tsx]: Good point. handle that in `css.tsx` file.

File: src/css/css.tsx [hunk 3]
```diff
@@ -592,6 +592,57 @@ const parseProps = {
...
+      const resolvedValue = variables.get(variableName);
+      if (resolvedValue !== undefined) {
+        return resolveVariables(resolvedValue, variables);
+      }
+      return fallback ? resolveVariables(fallback, variables) : '';
+    }
+  );
+}
+
+const propsToResolve = ['fill', 'color', 'stroke', 'stopColor', 'floodColor'];
```
Comment (jakex7) [inline src/css/css.tsx:634]: As per w3 specs:
`currentColor` applies to properties `fill`, `stroke`, `stop-color`, `flood-color` and `lighting-color` to I guess we should add `lightingColor` here

Comment (jakex7) [reply src/css/css.tsx:634]: https://www.w3.org/TR/SVG11/color.html
https://www.w3.org/TR/SVG11/filters.html#AnExample
```
<svg width="7.5cm" height="5cm" viewBox="0 0 200 120">
  <style>
    body {
      --my-color: #00f;
    }
  </style>
  <filter id="MyFilter" filterUnits="userSpaceOnUse" x="0" y="0" width="200" height="120">
    <feGaussianBlur in="SourceAlpha" stdDeviation="4" result="blur"/>
    <feOffset in="blur" dx="4" dy="4" result="offsetBlur"/>
    <feSpecularLighting in="blur" surfaceScale="5" specularConstant=".75" specularExponent="20" lighting-color="var(--my-color)" result="specOut">
      <fePointLight x="-5000" y="-10000" z="20000"/>
    </feSpecularLighting>
    <feComposite in="specOut" in2="SourceAlpha" operator="in" result="specOut"/>
    <feComposite in="SourceGraphic" in2="specOut" operator="arithmetic" k1="0" k2="1" k3="1" k4="0" result="litPaint"/>
    <feMerge>
      <feMergeNode in="offsetBlur"/>
      <feMergeNode in="litPaint"/>
    </feMerge>
  </filter>
  <rect x="1" y="1" width="198" height="118" fill="#888888" stroke="blue"/>
  <g filter="url(#MyFilter)">
    <g>
      <path fill="none" stroke="#D90000" stroke-width="10" d="M50,90 C0,90 0,30 50,30 L150,30 C200,30 200,90 150,90 z"/>
      <path fill="#D90000" d="M60,80 C30,80 30,40 60,40 L140,40 C170,40 170,80 140,80 z"/>
      <g fill="#FFFFFF" stroke="black" font-size="45" font-family="Verdana">
        <text x="52" y="76">SVG</text>
      </g>
    </g>
  </g>
</svg>
```

**Comment by bohdanprog** [comment]
@jakex7 Can we merge that PR?

**Comment by jakex7** [comment]
> @jakex7 Can we merge that PR?

Yes

**Pull request merged**